p2p/protocol/identify/id.go

```diff
@@ -139,31 +168,25 @@ func NewIDService(h host.Host, opts ...Option) (*idService, error) {
 		protocolVersion = cfg.protocolVersion
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
 	s := &idService{
-		Host:            h,
-		UserAgent:       userAgent,
-		ProtocolVersion: protocolVersion,
-
-		conns: make(map[network.Conn]chan struct{}),
-
+		Host:                    h,
+		UserAgent:               userAgent,
+		ProtocolVersion:         protocolVersion,
+		ctx:                     ctx,
```

Comment:
We have a `Close` method. It calls the cancel function of this context. Not sure if I like this pattern, but the alternative is to have `closeChan` that is closed in `Close` and a `sync.Once` that protects against duplicate calls to `Close` (closing a closed channel would panic).

Reply:
Using sync.Once is roughly equivalent to closing a context in terms of synchronization primitives. A closeChan would be preferable than a ctx here since the ctx is only being used as convenience instead of carrying request-scoped values (what's the request?).

Reply:
Let's not fix that here. We're using that pattern everywhere. I'm ok making that switch for all of our services in a separate PR.